src/WebUI/WebUI.csproj

```diff
@@ -1,24 +1,31 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
+    <PropertyGroup>
+        <TargetFramework>netcoreapp3.1</TargetFramework>
+        <RootNamespace>Codidact.WebUI</RootNamespace>
```

Comment:
I understand this is larger than the scope of this pull request, but the naming here doesn't follow the recommended pattern of Company.Technology.*.  In this case it should be Codidact.Web.UI.

This should be addressed in the coding standards when they get published and ratified.  We should fix this as early on in the project as possible, as addressing it later will be more difficult.

Reply:
>  We should fix this as early on in the project as possible, as addressing it later will be more difficult.

I agree, we should address this very quickly. It might already be difficult because we have a few large pull requests.

I'll look for some other locations and create an issue about it.

Reply:
@DoctaJonez I was going through the sources and must say, I am conflicted about whether this is a good idea or not.

Your explicit suggestion doesn't really fit the schema either, because `Web` isn't really a technology that we develop. I'd say we are creating an `QA` product, thus `Codidact.QA.WebServer` or something of the sort.

If this schema were applied rigorously it could look like this:

    Codidact.QA.WebServer
    Codidact.QA.Application
    Codidact.QA.Domain
    Codidact.QA.Infrastructure
    Codidact.QA.AuthServer

This looks better than before but only marginally. Unless there is another big plus that I missed, this doesn't seem to be worth the trouble.

Reply:
> Your explicit suggestion doesn't really fit the schema either, because `Web` isn't really a technology that we develop. I'd say we are creating an `QA` product, thus `Codidact.QA.WebServer` would be correct, or something of the sort.

WebServer is the wrong word because its not a server for the web. WebUI refers to it being a presentation layer, as in displays stuff in the web.

Also the other projects will be sharing the infrastructure, domain and application projects so when we have an API for instance it will access these projects. So you can't really relate these projects to only QA.

I like this answer personally: https://stackoverflow.com/a/3929357/2490286

Reply:
So you consider e.g. Infrastructure a project standing on its own? I am confused now.

If we were to add say a blog at `codidact.com/blog` that is completely independent from the question/answer thing, would these projects have the same infrastructure?

Reply:
> So you consider e.g. Infrastructure a project standing on its own? I am confused now.
> 
> If we were to add say a blog at `codidact.com/blog` that is completely independent from the question/answer thing, would these projects have the same infrastructure?

Actually giving it a lot of though these past 10 minutes I agree maybe we should add .QA after the Codidact since if we add a solution like blogging it will want to have its own infrastructure and domain.

Reply:
We might want to choose a different name though, there was some discussion somewhere about other types of posts, not just questions. I can't think of an umbrella term for it.

Reply:
Some good thoughts here.

I agree we need to choose an appropriate product name, which would leave the namespace as Codidact.ProductName.*

I had the same thoughts about Codidact.Web.* but for other reasons.  Codidact.Web is a valid namespace that we might want to use at some point.  Whenever we make utilities or extensions for various Microsoft namespaces, the convention is to mirror Microsoft's namespace, but using our own organisation name.  So we may end up with libraries along the lines of:

-Codidact.Web
-Codidact.AspNet.WebApi
-Codidact.Net.Http

I'm not saying we'd definitely need to implement these, just that it would be an appropriate convention, and we don't want to muddle this with application specific code.

These theoretical libraries could also be exposed as public nuget packages if we feel there's anything useful in there that's worth making available for general consumption.